### PR TITLE
[CDSR-1661] Move the callback from upscan handling to service layer

### PR DESCRIPTION
**Pull request by MJCallahanPage**

### CODE REVIEW

File: app/uk/gov/hmrc/uploaddocuments/services/FileUploadService.scala [hunk 1]
```diff
@@ -87,4 +90,67 @@ class FileUploadService @Inject()(repo: JourneyCacheRepository)
...
+                  uploadDetails.uploadTimestamp,
+                  uploadDetails.checksum,
+                  FileUpload.sanitizeFileName(uploadDetails.fileName),
+                  uploadDetails.fileMimeType,
+                  uploadDetails.size,
+                  description = context.config.newFileDescription
+                )
+              }
+          case UpscanFileFailed(_, failureDetails) =>
+            // update status of the file with matching nonce
```
Comment (frobian21) [inline app/uk/gov/hmrc/uploaddocuments/services/FileUploadService.scala:149]: move this comment to the top of the match case or just remove it

**Review by frobian21 — APPROVED** [review]

**Comment by MJCallahanPage** [comment]
☝️ rebased from main following PR merge

**Comment by frobian21** [comment]
I would say be careful with the separation of concern here - because we've stuck the connector into this service. I had an item to suggest the `FileUploadService` be moved to the repository layer as it is just doing our database operations - or not.
there's another service class in this repo which pulls in a connector and this fileUploadService and then mocks the calls to both.
As I was typing you updated the pr so this might not be relevant

**Comment by frobian21** [comment]
Nice I think the code is more readable so the comments were unnecessary - you'll have to  see what a fresh pair of eyes thinks

**Review by frobian21 — APPROVED** [review]

**Pull request merged**